src/components/EmojiPicker/EmojiPickerMenu/index.native.js

```diff
@@ -63,14 +64,47 @@ class EmojiPickerMenu extends Component {
         this.renderItem = this.renderItem.bind(this);
         this.isMobileLandscape = this.isMobileLandscape.bind(this);
         this.updatePreferredSkinTone = this.updatePreferredSkinTone.bind(this);
+        this.filterEmojis = _.debounce(this.filterEmojis.bind(this), 300);
         this.scrollToHeader = this.scrollToHeader.bind(this);
         this.getItemLayout = this.getItemLayout.bind(this);
+
+        this.state = {
+            filteredEmojis: this.emojis,
+            value: '',
+        };
     }
 
     getItemLayout(data, index) {
         return {length: CONST.EMOJI_PICKER_ITEM_HEIGHT, offset: CONST.EMOJI_PICKER_ITEM_HEIGHT * index, index};
     }
 
+    /**
+     * Filter the entire list of emojis to only emojis that have the search term in their keywords
+     *
+     * @param {String} searchTerm
+     */
+    filterEmojis(searchTerm) {
+        const normalizedSearchTerm = searchTerm.toLowerCase().trim();
```

Comment:
Since 99% of this is the same as `EmojiPickerMenu/index.js` I think we should extract it to a utils file.

Reply:
There is a difference with the onScroll logic. There are a lot of things that don't need to be handled in the native file. Also the difference in focus, arrow management, and a bunch of other things. However, if you want me to combine them, I'll do this

Reply:
Makes sense to me! Thanks for the context. Let's leave them separate.